### ISSUE TITLE
Implement another voting system : majority judgement

### ISSUE THREAD

**Issue by pletelli** [nextcloud/polls#3472]
### ⚠️ This issue respects the following points: ⚠️

- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Describe the goal you'd like to achieve

As a **poll participant**, I want to be able to vote with non binary answers because it helps to  respond with subtleties and to bring out a more consensual result.
As a **poll owner**, I want to **be able to** propose poll participant more than Yes, No, Maybe answers, and define the possible answer at each poll creation.
(excellent, good, fair, poor, awful) might be the default set of answers.

### Additional context

https://en.wikipedia.org/wiki/Majority_judgment

**Comment by septatrix** [comment]
While more voting variants would be pretty cool, this opens a whole new can of worms. There are several options (approval voting (yes/no), score voting (good, neutral, poor - or 1-10), ranking (order options by preference). Sadly none of them are ideal nor are the methods to evaluate them (first past the post, instant runoff, plurality, ...).
So there are many possible things which could be implemented here...

**Comment by artlog** [comment]
I have very simple server side code (toy code) done on purpose without relying on existing project, still should be integrated with front improvements and glue with poll database.

Another way would be to integrate existing work like Majority Judgement opensource code from https://mieuxvoter.fr. This could be a good ground if api and or front-end can be bundled & made compatible with nextcloud poll.
GitHub for those various project is at https://github.com/MieuxVoter .

**Comment by artlog** [comment]
For front part i first will do a poc using https://github.com/vinimoz/polls branch Rank_votes_feature from @vinimoz see comments in #3897.
I tested it on a docker nextcloud 31 (using podman).

![Image](https://github.com/user-attachments/assets/dc697978-fcbb-41c9-abff-a59dc1e452b0)

**Comment by vinimoz** [comment]
Thanks for the link https://mieuxvoter.fr/ , it's very interesting, i even wasn't aware and i'm french, such disaster :) Regarding this feature, if u guys wish, i can try to upgrade to version in vue3.

**Comment by artlog** [comment]
@vinimoz thanks for proposal of vue3 upgrade, if you have time it would be useful, if you think it is easy for you, i did try, not very hard, but it wasn't easy for me ;-)

**Comment by artlog** [comment]
I plan to pick this implementation for backend https://github.com/MieuxVoter/majority-judgment-library-php 
There is an interface PollTallyInterface with getParticipantsAmount()  and  getProposalsTallies() that may be mapped on database backend.

**Comment by vinimoz** [comment]
Good idea for the interface. I've myselft almost finish to implement my previous modification to vue3, just design and texting left, my previous modification to vue3.

**Comment by vinimoz** [comment]
<img width="1593" height="568" alt="Image" src="https://github.com/user-attachments/assets/08f472b4-c349-4065-93e7-f5569e71701b" />

**Comment by dartcafe** [comment]
I hope it is is just a mock-up. If you already implemented something I hardly recommend, you open a new PR starting from this repo and give a chance to review the steps you made. 

We will have to look into the architecture and the UX.

**Comment by vinimoz** [comment]
I’ve now submitted a clean pull request based on 8.0.6  from the official repo . The branch is `rank-vote-feature-vue3` and contains only the relevant changes for the ranked voting feature I implemented. https://github.com/vinimoz/polls/tree/rank-vote-feature-vue3
